### DISCUSSION TITLE
how to remove Assigning Domains from my vercel project

### DISCUSSION THREAD

**Discussion by AbdalrhmanHamza** [vercel/community#1220] [Help] [answered]
### Summary

I want to remove Assigning Domains from my vercel project because it affects my website's SEO

**Comment by swarnava** [comment, marked as answer]
Hi,

Incase you are trying to remove `vercel.app` domain, it's not possible to remove them. Thanks!

**Reply by amyegan** [reply]
Some additional info that might be helpful:
- Vercel automatically applies [Generated URLs](https://vercel.com/docs/concepts/deployments/generated-urls#) to all deployments
- [`x-robots-tag: noindex` is automatically applied to Preview deployments](https://vercel.com/docs/concepts/edge-network/headers#x-robots-tag) to avoid search engines crawling them
- You can designate a [canonical URL](https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls) to let search engines know which URL is the "real" one. Google offers a few different [ways to specify a canonical page](https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls#define-canonical)

**Reply by kylemclaren** [reply]
Hi! This needs further clarification, I think, as it's not 100% clear here or elsewhere this has been discussed. I would like to confirm that:

1. It is **not possible** to remove the [generated URL's](https://vercel.com/docs/deployments/generated-urls) like `<project-name>-<unique-hash>-<scope-slug>.vercel.app` from an app?
2. It is not possible to have a private app (protected with Cloudflare Access, for example) without paying for the $150 advanced deployment protection and all apps will always be publicly accessible without that add-on?

Any other (free) alternatives?

**Reply by amyegan** [reply]
Hey @kylemclaren. For your first question, the answer is yes. It is not possible to remove the generated URLs.

For your second question, there are many options to keep an app private. You already know about [Advanced Deployment Protection](https://vercel.com/docs/security/deployment-protection#advanced-deployment-protection). But my favorite option for a free solution is [Auth.js](https://authjs.dev/) with a sign in page in front of the app. It just takes a bit of setup and maintenance work. There are also other free tools and libraries that can accomplish the same thing.
